src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java

```diff
@@ -18,77 +17,119 @@
+	    waitForElementByElement(suggestionsList.get(0));		
+	}
+	
+	public void verifySuggestions(String suggestion) {
+		waitForElementByElement(suggestionsList.get(0));
+		String allSuggestions = "";
+		for (int i = 0; i < suggestionsList.size(); i++) {
+			allSuggestions += suggestionsList.get(i).getAttribute("title");
+		}
+		Assertion.assertStringContains(suggestion, allSuggestions);
+	}
 
-    public IntraWikiSearchPageObject searchFor(String query) {
-        searchInput.sendKeys(query);
-        searchSubmit.click();
-        PageObjectLogging.log("searchFor", "searching for query: " + query, true, driver);
-        return new IntraWikiSearchPageObject(driver);
-    }
+	/**
+    * Arrow down through suggestions, and click enter on the desired one
+	*/
+	public ArticlePageObject ArrowDownAndEnterSuggestion(String suggestion) {
+		waitForElementByElement(suggestionsList.get(0));
+		for (int i = 0; i < suggestionsList.size(); i++) {
+			WebElement currentSuggestion = suggestionsList.get(i);
+			searchInput.sendKeys(Keys.ARROW_DOWN);
+			if (currentSuggestion.getText().contains(suggestion)) {
+				searchInput.sendKeys(Keys.ENTER);
+				PageObjectLogging.log("ArrowDownToSuggestion", 
+				        "arrowed down to desired suggestion" +suggestion+"and clicked enter",
+				        true);
+				return new ArticlePageObject(driver);
+			}
+			else {
+				searchInput.sendKeys(Keys.ARROW_DOWN);
+			}
+		}
+		PageObjectLogging.log("ArrowDownToSuggestion", "didn't find suggestion: "+suggestion, false);
+		return null;
+	}
+	
+	/**
+    * click on desired suggestion 
+	*/
+	public ArticlePageObject clickSuggestion(String suggestion) {
+		waitForElementByElement(suggestionsList.get(0));
+		for (int i = 0; i < suggestionsList.size(); i++) {
+			WebElement currentSuggestion = suggestionsList.get(i);
+			if (currentSuggestion.getText().contains(suggestion)) {
+				currentSuggestion.click();
+				PageObjectLogging.log("clickSuggestion", 
+				        "clicked on desired suggestion" +suggestion,
+				        true);
+				return new ArticlePageObject(driver);
+			}
+		}
+		PageObjectLogging.log("clickSuggestion", "didn't find suggestion: "+suggestion, false);
+		return null;
+	}	
+	
+	public void typeQuery(String query) {
+		waitForElementByElement(searchInput);
+		searchInput.clear();
+		searchInput.sendKeys(query);
+		PageObjectLogging.log("typeQuery", "typed query: " + query, true);
+	}
 
-    /**
-     * Returns article page object if invoked by user with goSearch preference turned on
-     */
-    public ArticlePageObject goSearchFor(String query) {
-        searchInput.sendKeys(query);
-        searchSubmit.click();
-        PageObjectLogging.log("searchFor", "searching for query: " + query, true, driver);
-        return new ArticlePageObject(driver);
-    }
+	public IntraWikiSearchPageObject searchFor(String query) {
+		typeQuery(query);
+	    PageObjectLogging.log("searchFor", "searching for query: " + query, true, driver);
```

Comment:
yes but it was logged type query method being executed. 
I will reorder it so typeQuery is logged after searchFor is logged